LandSandBoat/server
# Feature/blu scadoo - BLU Spells to 99

**_I affirm:_**

I'm still in the middle of testing, but since this is a large PR I figured I'd open in now for anyone else that wants to check it out.

## What does this pull request do?

- [x] Adds all missing BLU spells
- [x] Refactors BLU Traits with values that follow along with retail 
- [x] Creates LUAs for all BLU spells not yet implemented with information from bg-wiki where available
- [x] Implements BLU JP Gift for trait bonuses 
- [x] Audited BLU spells for status effect tick minimum requirement of 3 

## Important notes
- **Fantod** is not properly implemented. It just copies MNK boost at the moment.
- Values not available from bg-wiki are either best guessed based on other spells or defaulted to basic values
- BLU Breath spells need another look at since they are HP based and it looks like the calculations aren't based on that. Most later breath spells are just a copy of Heat Breath.
- Animations for missing BLU spells are all set to 2 seconds and may need auditing.
- When increasing BLU job points, current active traits are not refreshed and require a job change or spell assignment to gain the newly received gift.

## Steps to test these changes
- Change BLU spells and verify traits are properly given.
- Cast BLU spells and verify effects, animation, and messaging.
- Check trait values when under the effect of any BLU JP Gift.
- Check spell **Magic Barrier** gives new magic stoneskin type effect.
